### PR TITLE
Add count of portfolio and refactor to improve readability

### PR DESCRIPTION
**Pull request by simpulton**

### CODE REVIEW

**Comment by simpulton** [comment]
> hooray! awesome sauce! 
does not tell me anything about the change.  please use a description of the change next time.

**Comment by simpulton** [comment]
For example data in tests, I think we could extract-out some of the client data:
* generation where the specifics don't matter
* have a more-varied data generation approach to catch edge cases

File: src/app/common/reducers/clients.reducer.ts [hunk 1]
```diff
@@ -15,22 +15,34 @@ export const initialState: State = {
...
-        });
-      }, {});
-
-      return {
-        ids,
-        entities,
-        selectedClientId: state.selectedClientId
-      };
+      return transformClients(state, action);
     case actions.ActionTypes.SELECT:
```
Comment (simpulton) [inline src/app/common/reducers/clients.reducer.ts]: let's extract these in a subsequent commit / PR

**Review by simpulton — COMMENTED** [review]
approved by @skuenzli
